### DISCUSSION TITLE
Snippet in Form not being translated

### DISCUSSION THREAD

**Discussion by enzedonline** [wagtail/wagtail-localize#384] [Q&A] [answered]
Not sure if this is a bug or if I just haven't coded this right.

I have email signatures set up as translatable snippets. These are set as a field in my custom abstract email form.
The snippets are sync'd and translated ok. The forms are translating fine except for the signatures. 
I'm referencing the signature using localized (`self.receipt_email_footer.localized.signature_content`) but this seems to have no effect.
When I'm in the translating page for the form (image from the French translation page), the signature shows the default signature, there is only the option to edit the translated snippet.
![image](https://user-images.githubusercontent.com/72310199/116819950-d7330a80-ab72-11eb-8949-4a2ee4f14ba4.png)

How would I go about retrieving the translated snippet?

Code below:
```
@register_snippet
class EmailSignature(TranslatableMixin, models.Model):
    signature_name = models.CharField(
        max_length=30,
        null=False,
        blank=False,
        verbose_name=_("Email Signature Title"),
        help_text=_("Used to identify this signature")
    )
    signature_content = RichTextField(
        features= [
            'h1', 'h2', 'h3', 'h4', 'h5', 'h6',
            'bold',
            'italic',
            'ol',
            'ul',
            'link',
            'hr',
            'center',
        ],
        verbose_name=_("Email Signature Content"),
        help_text=_("Text for the Email Signature")
    )
    signature_image = models.ForeignKey(
        'wagtailimages.Image',
        blank=True,
        null=True,
        related_name='+',
        on_delete=models.SET_NULL,
        verbose_name=_("Image for Left Column"),
        help_text=_("Image to display in left column of Email Signature")
    )

    panels = [
        FieldPanel('signature_name'),
        RichTextFieldPanel('signature_content'),
        ImageChooserPanel('signature_image'),
    ]

    def __str__(self):
        """The string representation of this class"""
        return self.signature_name

    class Meta:
        verbose_name = _('Email Signature')
        verbose_name_plural = _('Email Signatures')
        unique_together = ('translation_key', 'locale')
```

In the form page class:
```
    receipt_email_footer = models.ForeignKey(
        'site_settings.EmailSignature',
        null=True,
        blank=True,
        on_delete=models.SET_NULL,
        related_name='+'
    )
```

In the send mail method:
`footer = self.receipt_email_footer.localized.signature_content`

Have also tried
`locale_footer = EmailSignature.objects.get(id=self.receipt_email_footer.id).localized`
then referencing `locale_footer.signature_content`, but I still get the default language signature.

Thanks again for any assistance.

*edit*
As a hack, I put the translated text into the db directly (the relevant field in site_settings_emailsignature), it picks that up ok, just not the translated text. Seems it doesn't recognise the translation for that model.

**Comment by enzedonline** [comment, marked as answer]
I still haven't got to the bottom of this. The email signature is definitely the localized version, but the translated value is ignored, the value in the page table is ready instead:

```
>>> pages=ContactPage.objects.all()
>>> enpg=pages.first()
>>> translation.activate('es')
>>> enpg
<ContactPage: Contact Me>
>>> espg=enpg.localized
>>> espg
<ContactPage: Contáctame>
>>> enpg.receipt_email_footer
<EmailSignature: Default Signature (English)>
>>> enpg.receipt_email_footer.id
1
>>> espg.receipt_email_footer
<EmailSignature: Default Signature (English)>
>>> espg.receipt_email_footer.id
2

>>> trans=enpg.receipt_email_footer.get_translations() 
>>> trans
<QuerySet [<EmailSignature: Default Signature (English)>, <EmailSignature: Default Signature (French)>]>
```

In the QuerySet, the first value is the Spanish signature, the second, the French, I put the value directly into the database emailsignature table.

Translation page for the signature:
![image](https://user-images.githubusercontent.com/72310199/117083578-50825700-ad45-11eb-8222-e2966f549314.png)

Would really appreciate any feedback on this one ...

**Reply by kaedroho** [reply]
Wagtail localize should automatically set the foreign key on the translated page to point at the translated snippet.

I'm not sure why the text is incorrect though, it might be that you do need to "publish" the snippet when you edit it with Wagtail Localize (since the translations that are shown in the editor are actually stored somewhere else and not written to the snippet until publish).

**Reply by enzedonline** [reply]
Hi Karl,

It's got me stumped as well.

There's only the option to publish when saving the translation on a snippet here:
![image](https://user-images.githubusercontent.com/72310199/117307096-91cb5180-ae80-11eb-85b2-1aae70fc7803.png)

**Reply by enzedonline** [reply]
I'll close this - works on the prod server but not the dev server despite being identical.
